### PR TITLE
[SPARK-11595] [SQL] [BRANCH-1.5] Fixes ADD JAR when the input path contains URL scheme 

### PR DESCRIPTION
**Pull request by liancheng**
This PR backports #9569 to branch-1.5.

### CODE REVIEW

**Comment by yhuai** [comment]
Is the failed test a legitimate one?

**Comment by liancheng** [comment]
The failure is because of another bug that has been fixed in master by PR #9277. Will update the test case here and see whether we should backport #9277 to branch-1.5.

**Comment by yhuai** [comment]
Let's add this test in JDBC tests.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala [hunk 1]
```diff
@@ -70,7 +70,12 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils {
   private val sqlContext = _sqlContext
 
   test("UDTF") {
-    sql(s"ADD JAR ${TestHive.getHiveFile("TestUDTF.jar").getCanonicalPath()}")
+    val jarPath = TestHive.getHiveFile("TestUDTF.jar").getCanonicalPath
+
+    // SPARK-11595 Fixes ADD JAR when input path contains URL scheme
+    val jarURL = s"file://$jarPath"
```
Comment (yhuai) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]: Should we also check a URL without a scheme?

Comment (liancheng) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]: Is it possible to remove an added jar? I'm afraid if we add the other test case, these two may interfere with each other.

Comment (liancheng) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]: Actually there are other test cases that already covered the case without a scheme. So I think it's fine.

Comment (yhuai) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]: I think it is not possible.

Comment (yhuai) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]: ok

**Comment by liancheng** [comment]
I've added the Thrift server test case in #9664. Thrift test cases are fairly slow. Would like to combine them.

#9664 is for master branch, opening backport branch for branch-1.5 soon.

**Comment by yhuai** [comment]
Thanks! Merging to branch 1.5.

**Comment by yhuai** [comment]
I have merged it to branch 1.5.

**Pull request closed without merging**